### PR TITLE
Add more features to serializer

### PR DESCRIPTION
**Pull request by runeh**
- make `export` prop optional on root type
- add support for readonly/nullable/both/none members on `Record`
- add support for symbol

### CODE REVIEW

File: src/main_alt.ts [hunk 1]
```diff
@@ -97,27 +98,72 @@ function writeUnionType(w: CodeBlockWriter, node: UnionType) {
...
     writeAnyType(w, type);
-    w.write(', ');
+    w.write(',\n');
   }
-  w.write(') ');
+  w.writeLine(') ');
+}
+
+/**
+ * public for testing
```
Comment (cobraz) [inline src/main_alt.ts:116]:
```suggestion
 * public for testing
 *
 * Used to evaluate if node.fields include `readonly` and/or `nullable`
```

File: src/main_alt.ts [hunk 2]
```diff
@@ -97,27 +98,72 @@ function writeUnionType(w: CodeBlockWriter, node: UnionType) {
...
+    w.writeLine('rt.Record({');
+    for (const field of fieldKind.fields) {
+      w.write(field.name);
+      w.write(': ');
+      writeAnyType(w, field.type);
+      w.write(',\n');
+    }
+    w.write('})');
+    w.conditionalWrite(fieldKind.nullable ?? false, '.asPartial()');
+    w.conditionalWrite(fieldKind.readonly ?? false, '.asReadonly()');
```
Comment (cobraz) [inline src/main_alt.ts:165]: Nice :)

So `groupFieldKinds` is used for these conditions. They are evaluated based on the `node.fields` and if there are `readonly` and/or `nullable` fields.

**Review by cobraz — APPROVED** [review]
Thank you so much, this is looking very nice!

File: src/main_alt.ts (hunk 2 above)
Comment (runeh) [reply src/main_alt.ts:167]: Yup. No other way, right now, to make a `Record` where some stuff is partial and some is nullable. You can use `.And(` but that's just an intersection with different syntax. There's new stuff that just landed in runtypes that can  be used as well when it's released: https://github.com/pelotom/runtypes#optional-values

**Pull request merged**